website/content/docs/security/acl/auth-methods/aws-iam.mdx

```diff
@@ -0,0 +1,199 @@
+- `AllowedSTSHeaderValues` `(array<string>: [])` - A list of additional allowed headers on
+  `sts:GetCallerIdentity` requests. In any case, a default list of headers AWS STS expects are
+  allowed.
+
+### Sample
+
+```json
+{
+    ...other fields...
+    "Config": {
+      "BoundIAMPrincipalARNs": ["arn:aws:iam::123456789012:role/MyRoleName"],
+      "EnableIAMEntityDetails": true,
+      "IAMEntityTags": ["consul-namespace"],
+      "ServerIDHeaderValue": "my.consul.server.example.com",
+      "MaxRetries": 3,
+      "IAMEndpoint": "https://iam.amazonaws.com/",
+      "STSEndpoint": "https://sts.us-east-1.amazonaws.com/",
+      "AllowedSTSHeaderValues": ["X-Extra-Header"]
+    }
+}
+```
+
+## Trusted Identity Attributes
+
+The authentication step returns the following trusted identity attributes for use in binding rule
+selectors and bind name interpolation. All of these attributes are strings that can be interpolated
+and support the following selector operations: `Equal, Not Equal, In, Not In, Matches, Not Matches`
+
+| Attribute            | Description                         | Requirement                                                      |
+| -------------------- | ----------------------------------- | ---------------------------------------------------------------- |
+| `entity_name`        | Name of IAM role or user            |                                                                  |
+| `entity_id`          | Unique ID of IAM role or user       |                                                                  |
+| `account_id`         | AWS account id of IAM role or user  |                                                                  |
+| `entity_path`        | The path of the IAM role or user    | `EnableIAMEntityDetails=true`                                    |
+| `entity_tags.<key>`  | AWS account id of IAM role or user  | `EnableIAMEntityDetails=true` and `IAMEntityTags` contains `<key>` |
+
+## IAM Policies
+
+When `EnableIAMEntityDetails=false`, no specific IAM policies are needed.
+
+When `EnableIAMEntityDetails=true`, an authenticating client must provide either a signed
+`iam:GetRole` or `iam:GetUser` request. This request is signed with the client's AWS credentials, so
+the client must have permission to fetch the role or user, respectively.
+
+- If the authenticating client is an IAM role, the client must have an `iam:GetRole` permission to
+  fetch its own role. The following shows an example of an IAM Policy document which grants this
+  permission:
+
+  ```json
+  {
+    "Statement": [
+        {
+            "Action": ["iam:GetRole"],
+            "Effect": "Allow",
+            "Resource": ["arn:aws:iam::123456789012:role/MyRoleName"]
+        }
+    ],
+    "Version": "2012-10-17"
+  }
+  ```
+
+- If the authenticating client is an IAM user, the client must have an `iam:GetUser` permission to
+  fetch its own role. The following shows an example of an IAM Policy document which grants this
```

Comment:
```suggestion
  fetch its own role. The following shows an example of an AWS IAM Policy document which grants this
```
It might be silly to add AWS here, but because we are jumping back and forth between Consul config and AWS IAM policy it's better to be explicit to prevent the potential for confusion.